carousel/core/models.py

```diff
@@ -85,35 +86,18 @@ class Model(object):
     def __init__(self, modelfile=None):
         meta = getattr(self, ModelBase._meta_attr)
         parameters = getattr(self, ModelBase._param_attr)
-
-        # # make model from parameters
-        # self.model = dict.fromkeys(meta.layer_cls_names)
-
         # load modelfile if it's an argument
         if modelfile is not None:
             #: model file
             self.param_file = os.path.abspath(modelfile)
             LOGGER.debug('modelfile: %s', modelfile)
         else:
             modelfile = self.param_file
-        # =====================================================================
-        # TODO: don't monkey patch _meta, it breaks tests with same class
-        if not hasattr(meta, 'modelpath'):
-            # modelfile was either given as arg or wasn't in metaclass
-            modelpath = None  # modelpath will be derived from modelfile
-        else:
-            modelpath = meta.modelpath
-        # get modelpath from modelfile if not in meta class
-        if modelfile is not None and modelpath is None:
-            #: model path, used to find layer files relative to model
-            modelpath = os.path.dirname(os.path.dirname(modelfile))
-            # meta.modelpath = modelpath
-        # =====================================================================
         # check meta class for model if declared inline
         if parameters:
             # TODO: separate model and parameters according to comments in #78
             #: dictionary of the model
-            self.model = model = parameters
+            self.model = model = copy.deepcopy(parameters)
```

Comment:
no, this is a temporary maybe even pointless hack. I don't want to lose the parameters if they've already been imported as `Parameter` but the way carousel currently works, it needs to deserialize the model into a JSON string, and then re-import the layers and their parameters. Silly. But I can't couldn't change that in this PR, so I made some issues like #88 and [this comment](../issues/78#issuecomment-260109368) in #78